### PR TITLE
[PKG-1913] pyppmd >=0.18.1,<1.1.0

### PR DESCRIPTION
**Pull request by rafaelmartins**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -37,10 +38,17 @@ test:
...
-  home: http://github.com/miurahr/pyppmd
+  description: |
+    Classes and functions for compressing and decompressing text data, using
+    PPM(Prediction by partial matching) compression algorithm which has several
+    variations of implementations.
+  home: https://codeberg.org/miurahr/pyppmd
+  dev_url: https://codeberg.org/miurahr/pyppmd
+  doc_url: https://codeberg.org/miurahr/pyppmd/src/branch/main/README.rst
   summary: PPMd compression/decompression library
   license: BSD-3-Clause
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml]: license should be LGPLv2
see https://github.com/miurahr/pyppmd/blob/releases/1.0/LICENSE

File: recipe/meta.yaml [hunk 2]
```diff
@@ -37,10 +38,17 @@ test:
...
-  license_file: LICENSE.rst
+  license_family: BSD
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml]: LGPL

File: recipe/meta.yaml (hunk 1 above)
Comment (rafaelmartins) [reply recipe/meta.yaml]: I found another mistake, will close for now

**Comment by rafaelmartins** [comment]
@M-Waszkiewicz-Anaconda all fixed

File: recipe/meta.yaml [hunk 3]
```diff
@@ -22,11 +21,12 @@ requirements:
...
     - python
-    - setuptools >=45.0
-    - setuptools-scm >=6.0.1
+    - setuptools
+    - setuptools-scm 7.0.4
+    - wheel
   run:
     - python
-    - importlib_metadata
+    - importlib_metadata  # [py<38]
```
Comment (cbouss) [inline recipe/meta.yaml:29]:
```suggestion
```

I don't see it in https://codeberg.org/miurahr/pyppmd/src/tag/v1.0.0/pyproject.toml

**Review by cbouss — APPROVED** [review]
LGTM, with importlib_metadata removed.

File: recipe/meta.yaml (hunk 3 above)
Comment (rafaelmartins) [reply recipe/meta.yaml:29]: @cbouss that's an upstream mistake, it is not listed, but still used for python<3.8:
https://codeberg.org/miurahr/pyppmd/src/branch/main/src/pyppmd/__init__.py#L6

however, as we don't build for such old python versions, I think that we could remove. I kept it for consistency with other recipes.

Opinions?

Comment (cbouss) [reply recipe/meta.yaml:29]: Ok, keeping it is a good call then.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -22,11 +21,12 @@ requirements:
   host:
     - pip
     - python
-    - setuptools >=45.0
-    - setuptools-scm >=6.0.1
+    - setuptools
+    - setuptools-scm 7.0.4
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/meta.yaml:25]: why not 6.0.1 according to https://github.com/miurahr/pyppmd/blob/bbfd861dad4e9a983c102a9705eb8a192653c581/pyproject.toml#L64 ?

Comment (rafaelmartins) [reply recipe/meta.yaml:25]: @M-Waszkiewicz-Anaconda you are looking at old code, the new repository is https://codeberg.org/miurahr/pyppmd

The 1.0.0 release was still created with setuptools-scm 6.0.1, but it fails to build with `--no-deps --no-build-isolation`. The upstream developer fixed this after the release by increasing the requirement, but do not created a patch release: https://codeberg.org/miurahr/pyppmd/commit/7edc2d65467f1ec1e9a11828036b8a33cf336a37

7.0.4 is our built version that satisfies the requirements and everything works fine.

**Review by ELundby45 — APPROVED** [review]

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

File: recipe/meta.yaml (hunk 4 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:25]: Missing `toml`, see https://github.com/miurahr/pyppmd/blob/bbfd861dad4e9a983c102a9705eb8a192653c581/pyproject.toml#L64

Comment (rafaelmartins) [reply recipe/meta.yaml:25]: `toml`/`tomli` is already installed unconditionally by our `setuptools_scm` recipe. I see no need to include it in each package and polute the dependency graphs with sub-dependencies.

**Pull request merged**